core/src/commitment.rs

```diff
@@ -123,12 +131,12 @@ impl BlockCommitmentCache {
     }
 
     pub fn is_confirmed_rooted(&self, slot: Slot) -> bool {
-        self.get_block_commitment(slot)
-            .map(|block_commitment| {
-                (block_commitment.get_rooted_stake() as f64 / self.total_stake as f64)
-                    > VOTE_THRESHOLD_SIZE
-            })
-            .unwrap_or(false)
+        if let Some(block_commitment) = self.get_block_commitment(slot) {
+            (block_commitment.get_rooted_stake() as f64 / self.total_stake as f64)
+                > VOTE_THRESHOLD_SIZE
```

Comment:
hmmm, is this case covered by the `else` clause, aka is it true:

`(get_block_commitment(slot).get_rooted_stake() as f64 / self.total_stake > VOTE_THRESHOLD_SIZE` => `slot <= self.largest_confirmed_root()`

Reply:
To make sure this slot is a root (less than a root doesn't necessarily mean it's on the roote d path) we could also check blockstore.is_root()

Reply:
Oh duh, yes, it should be! I'll write a test case for it, and remove.

Reply:
> To make sure this slot is a root (less than a root doesn't necessarily mean it's on the roote d path) we could also check `blockstore.is_root(slot)`

I was placing that burden on the caller, but... give AggregateCommitmentService a ref to Blockstore, you think?

Reply:
hehe yeah I think so. So here's a question, let's say the chain of slots looks like:

2->4->6->8->10->12->....-> 18-> 20

Let's say we've rooted 6, 8 in our blockstore (our last vote was 14 let's say). But the highest confirmed root based on our above computations is 12. 

Now let's say someone comes in and asks about 10, should we say it's a root even though we haven't rooted it yet (we've observed 2/3 have rooted it though)? If so, I think we need to check `blockstore.is_root(slot) || ancestors(self. largest_confirmed_root).contains(slot)`

Reply:
Based on our earlier discussions on Slack, I think that we do want it to be rooted as per this node to say it's a confirmed root.

Reply:
yeah it's not critical I think, would just be nice to have. Some nodes that are slightly behind no voting may temporarily respond "not confirmed", but they should catch up eventually